### ISSUE TITLE
Payara Micro 5.184: unable to fire async event

### ISSUE THREAD

**Issue by mlvtito** [payara/Payara#3861]
# Description #
----------

Each time an event is asynchronously fired with CDI, I got an exception. This was working inversion 5.183.

We can reproduce the problem with this sample project https://github.com/mlvtito/issue-cdi-async

## Expected Outcome

I expect the event to be processed by observers methods.

In my sample program, I expect to see this message in log :

```
####################### Processing event FIRE ASYNC
```

## Current Outcome

This error occurs in log :

```
####################### Error while processing event
java.util.concurrent.CompletionException: java.lang.IllegalStateException: Singleton not set for com.sun.enterprise.loader.CurrentBeforeParentClassLoader@5eb2cb98
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1592)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: Singleton not set for com.sun.enterprise.loader.CurrentBeforeParentClassLoader@5eb2cb98
	at org.jboss.weld.bootstrap.api.helpers.TCCLSingletonProvider$TCCLSingleton.get(TCCLSingletonProvider.java:54)
	at org.jboss.weld.Container.instance(Container.java:65)
	at org.jboss.weld.event.ObserverNotifier.lambda$new$2(ObserverNotifier.java:113)
	at org.jboss.weld.util.LazyValueHolder$1.computeValue(LazyValueHolder.java:32)
	at org.jboss.weld.util.LazyValueHolder.get(LazyValueHolder.java:46)
	at org.jboss.weld.event.ObserverNotifier.lambda$createSupplier$9(ObserverNotifier.java:434)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
	... 3 more
```

## Steps to reproduce (Only for bug reports) 

1 -**  clone the sample project

    git clone https://github.com/mlvtito/issue-cdi-async.git

2 -**  build the project

    mvn clean install

3 -**  start payara micro

    mvn payara-micro:start 

4 -**  invoke http://localhost:8080/issue-cdi-async/api/test and watch the log

### Samples

https://github.com/mlvtito/issue-cdi-async

If you change property `version.payara.micro` in `pom.xml` file to 5.183. Everything is working fine.

## Environment ##

- **Payara Version**: 5.184
- **Edition**: Micro
- **JDK Version**: OpenJDK Runtime Environment (Zulu 8.36.0.5-CA-linux32) (build 1.8.0_202-b05)
- **Operating System**: Linux Ubuntu 18.04
- **Database**: No database

**Comment by smillidge** [comment]
Looks like the wrong classloader is ending up on the asynch thread.

**Comment by smillidge** [comment]
Fixed by #3880
